seisbench/data/base.py

```diff
@@ -442,6 +448,17 @@ def _get_component_mapping(self, source, target):
         :param target:
         :return:
         """
+
+        if (
+            isinstance(source, float)
+            and np.isnan(source)
+            or (
+                (isinstance(source, list) or isinstance(source, str))
+                and not len(source)
+            )
+        ):
+            raise ValueError(f"Component order not set for trace or dataset.")
```

Comment:
I think this is a good way of going about this. I'd just slightly rephrase the exception text, as this will not be called on specific traces but only once the dataset is loaded (or the target component order is changed, but the exception should never happen then). How about "Component order not set for (parts of) the dataset."?